server/src/services/data_puller.ts

```diff
@@ -82,7 +82,9 @@ export class DataPuller {
       throw Error(`data puller can't push unit: ${unit} data: ${data}`);
     }
     let task = new AnalyticsTask(unit.id, AnalyticsTaskType.PUSH, data);
-
+    if (data.cache.length == 0) {
```

Comment:
You're checking for an empty array here
But it could be an empty object or `null`
```suggestion
    if(_.isEmpty(data.cache) || data.cache === null) {
```